src/Config/ConfigLoader.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Robo\Config;
+
+use Grasmash\YamlExpander\Expander;
+
+/**
+ * Load configuration files, and fill in any property values that
+ * need to be expanded.
+ */
+class ConfigLoader implements ConfigLoaderInterface
+{
+    protected $config = [];
+    protected $source = '';
+
+    public function getSourceName()
+    {
+        return $this->source;
+    }
+
+    public function setSourceName($source)
+    {
+        $this->source = $source;
+        return $this;
+    }
+
+    public function import($data)
+    {
+        $this->config = [];
+        return $this->add($data);
+    }
+
+    public function add($data)
+    {
+        if ($data instanceof ConfigLoaderInterface) {
+            $data = $data->export();
+        }
+        $this->config = Expander::expandArrayProperties($data, $this->config);
```

Comment:
I'm not sure that I want to support expanding dot-notation keys. Instead of `bin.path = '/some/path'`:

```
bin:
  path: /some/path
```
Configuration values from different config files definitely should be merged, not overwritten , though. I will look into that.

Of course, if you are using Robo as a Framework, you can load configuration however you'd like. I would suggest making dot-notation key expansion a function of the yaml loader, not the config object.